src/rewriter/rewrite_db_proof_cons.cpp

```diff
@@ -518,11 +519,60 @@ bool RewriteDbProofCons::proveWithRule(RewriteProofStatus id,
   }
   else if (id == RewriteProofStatus::ARITH_POLY_NORM)
   {
-    if (!theory::arith::PolyNorm::isArithPolyNorm(target[0], target[1]))
+    if (target[0].getType().isBoolean())
     {
-      return false;
+      Rational rx, ry;
+      if (!theory::arith::PolyNorm::isArithPolyNormRel(
+              target[0], target[1], rx, ry))
+      {
+        return false;
+      }
+      NodeManager* nm = nodeManager();
+      Node lhs, rhs;
+      if (target[0][0].getType().isBitVector())
+      {
+        uint32_t wa = target[0][0].getType().getBitVectorSize();
+        uint32_t wb = target[1][0].getType().getBitVectorSize();
+        Node cx = nm->mkConst(BitVector(wa, rx.getNumerator()));
+        Node cy = nm->mkConst(BitVector(wb, ry.getNumerator()));
+        Node x = nm->mkNode(Kind::BITVECTOR_SUB, target[0][0], target[0][1]);
+        Node y = nm->mkNode(Kind::BITVECTOR_SUB, target[1][0], target[1][1]);
+        lhs = nm->mkNode(Kind::BITVECTOR_MULT, cx, x);
+        rhs = nm->mkNode(Kind::BITVECTOR_MULT, cy, y);
+      }
+      else
+      {
+        Node x = nm->mkNode(Kind::SUB, target[0][0], target[0][1]);
+        Node y = nm->mkNode(Kind::SUB, target[1][0], target[1][1]);
+        Node cx, cy;
+        if (x.getType().isInteger() && y.getType().isInteger())
```

Comment:
You can merge these two blocks by calling `nm->mkConstRealOrInt` with the type of `x` and `y`.